search/sublist_search.cpp

```diff
@@ -0,0 +1,338 @@
+        std::cout << std::endl;
+    }
+
+    /**
+     * @brief A test case contains edge case, Only contains one element.
+     * @returns void
+     * */
+    void testCase_1() {
+        const bool expectedOutput = true;
+
+        log("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+            "~");
+        log("This is test case 1 for sublist search Algorithm : ");
+        log("Description:");
+        log("   EDGE CASE : Only contains one element");
+
+        std::vector<uint64_t> sublistData = {6};
+        std::vector<uint64_t> mainlistData = {2, 5, 6, 7, 8};
+
+        search::sublist_search::Node *sublistLL =
+            search::sublist_search::makeLinkedList(sublistData);
+        search::sublist_search::Node *mainlistLL =
+            search::sublist_search::makeLinkedList(mainlistData);
+
+        bool exists =
+            search::sublist_search::sublistSearch(sublistLL, mainlistLL);
+
+        log("Checking assert expression...");
+        assert(exists == expectedOutput);
+        log("Assertion check passed!");
+
+        log("[PASS] : TEST CASE 1 PASS!");
+        log("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+            "~");
+
+        delete (sublistLL);
+        delete (mainlistLL);
+    }
+
+    /**
+     * @brief A test case which contains main list of 100 elements and sublist
+     * of 20.
+     * @returns void
+     * */
+    void testCase_2() {
+        const bool expectedOutput = true;
+
+        log("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+            "~");
+        log("This is test case 2 for sublist search Algorithm : ");
+        log("Description:");
+        log("   contains main list of 100 elements and sublist of 20");
+
+        std::vector<uint64_t> sublistData(20);
+        std::vector<uint64_t> mainlistData(100);
+
+        for (int i = 0; i < 100; i++) {
+            // Inserts 100 elements in main list
+            mainlistData[i] = i + 1;
+        }
+
+        int temp = 0;
+        for (int i = 45; i < 65; i++) {
+            // Inserts 20 elements in sublist
+            sublistData[temp] = i + 1;
+            temp++;
+        }
+
+        search::sublist_search::Node *sublistLL =
+            search::sublist_search::makeLinkedList(sublistData);
+        search::sublist_search::Node *mainlistLL =
+            search::sublist_search::makeLinkedList(mainlistData);
+
+        bool exists =
+            search::sublist_search::sublistSearch(sublistLL, mainlistLL);
+
+        log("Checking assert expression...");
+        assert(exists == expectedOutput);
+        log("Assertion check passed!");
+
+        log("[PASS] : TEST CASE 2 PASS!");
+        log("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+            "~");
+    }
+
+    /**
+     * @brief A test case which contains main list of 50 elements and sublist
+     * of 20.
+     * @returns void
+     * */
+    void testCase_3() {
```

Comment:
I'm not sure if these tests should be `static` too. @ayaankhan98, @mishraabhinn?

Reply:
In this particular case, there is no reason why it must be static (based on the code provided).

Reply:
In tests cases, we use `static` for a Doxygen bug. Although, the main test function is already `static`, so I'm not sure if to keep it as-is or make it static. Let's see if the documentation is fine after merging this PR. 🙂